overlord/overlord_test.go

```diff
@@ -176,7 +176,8 @@ func (ovs *overlordSuite) TestNewWithGoodState(c *C) {
 		"last-change-id": 0,
 		"last-task-id": 0,
 		"last-lane-id": 0,
-		"last-notice-id": 0
+		"last-notice-id": 0,
+		"notice-last-date":"0001-01-01T00:00:00Z"
```

Comment:
Here as well: `"notice-last-timestamp"` and nanosecond precision?

Reply:
AFAIK, nanosecond precision is really not needed if it's zero...

Reply:
Ok, tested, and confirmed that it removes trailing zeros, so an integer value (zero in this case) won't have "dot zero zero zero..."

Reply:
My thought was to test that everything works as expected if the timestamp has nonzero nanoseconds, since afaik the odds of the nanoseconds being zero in the real world are one in a billion :)

Am I missing something though, is there a reason the time will be integer-valued seconds?

Reply:
[Playground link](https://go.dev/play/p/LJ-5ojPjXbI) that demonstrates how Go's encoding/json removes trailing zeroes:

```
time.Date(2024, time.January, 2, 3, 4, 5, 0, time.UTC) -> "2024-01-02T03:04:05Z"
time.Date(2024, time.January, 2, 3, 4, 5, 100000000, time.UTC) -> "2024-01-02T03:04:05.1Z"
time.Date(2024, time.January, 2, 3, 4, 5, 120000000, time.UTC) -> "2024-01-02T03:04:05.12Z"
time.Date(2024, time.January, 2, 3, 4, 5, 123000000, time.UTC) -> "2024-01-02T03:04:05.123Z"
time.Date(2024, time.January, 2, 3, 4, 5, 123400000, time.UTC) -> "2024-01-02T03:04:05.1234Z"
time.Date(2024, time.January, 2, 3, 4, 5, 123450000, time.UTC) -> "2024-01-02T03:04:05.12345Z"
time.Date(2024, time.January, 2, 3, 4, 5, 123456000, time.UTC) -> "2024-01-02T03:04:05.123456Z"
time.Date(2024, time.January, 2, 3, 4, 5, 123456700, time.UTC) -> "2024-01-02T03:04:05.1234567Z"
time.Date(2024, time.January, 2, 3, 4, 5, 123456780, time.UTC) -> "2024-01-02T03:04:05.12345678Z"
time.Date(2024, time.January, 2, 3, 4, 5, 123456789, time.UTC) -> "2024-01-02T03:04:05.123456789Z"
```

So the JSON-encoded zero value of time.Time will not have nanoseconds. It looks like we do use nanoseconds in the test, so I think this is fine.

Reply:
Apologies, I didn't really process that this was encoding the zero-value for the time type, not an explicitly-specified time, so I see why we can't have nanosecond precision here. And the new `TestAvoidTwoNoticesWithSameDateTime` checks json marshalling/unmarshalling with nonzero nanoseconds. So all good from me, thanks for your patience :)